src/components/Validations/Validation.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { FunctionComponent } from 'react';
+import { ObjectCheck, ObjectValidation } from '../../types/IstioObjects';
+import { ErrorCircleOIcon, OkIcon, WarningTriangleIcon } from '@patternfly/react-icons';
+import { PfColors } from '../Pf/PfColors';
+import { IconProps } from '@patternfly/react-core';
+
+type Props = {
+  validation?: ObjectValidation;
+};
+
+enum ValidationTypes {
+  Error = 'error',
+  Warning = 'warning',
+  Correct = 'correct'
+}
+
+type ValidationType = {
+  color: string;
+  icon: FunctionComponent<IconProps>;
+};
+
+const ErrorValidation: ValidationType = {
+  color: PfColors.Red100,
+  icon: ErrorCircleOIcon
+};
+
+const WarningValidation: ValidationType = {
+  color: PfColors.Orange400,
+  icon: WarningTriangleIcon
+};
+
+const CorrectValidation: ValidationType = {
+  color: PfColors.Green400,
+  icon: OkIcon
+};
+
+class Validation extends React.Component<Props> {
+  isValid(): boolean {
+    return !!this.props.validation && this.props.validation.valid;
+  }
+
+  numberOfChecks(type: string): number {
+    const object = this.props.validation;
+    let count = 0;
+
+    if (object) {
+      count = (object && object.checks ? object.checks : []).filter(i => i.severity === type).length;
+    }
+
+    return count;
+  }
+
+  validation(): ValidationType {
+    if (!this.props.validation) {
+      return ErrorValidation;
+    }
+
+    const object = this.props.validation;
+    const warnChecks = this.numberOfChecks(ValidationTypes.Warning);
+    const errChecks = this.numberOfChecks(ValidationTypes.Error);
+
+    let validation: ValidationType = CorrectValidation;
+    if (!object.valid) {
+      if (errChecks > 0) {
+        validation = ErrorValidation;
+      } else if (warnChecks > 0) {
+        validation = WarningValidation;
+      }
+    }
+
+    return validation;
+  }
+
+  checkForPath(path: string): ObjectCheck[] {
+    const object = this.props.validation;
+
+    if (!object || !object.checks) {
+      return [];
+    }
+
+    const check = object.checks.filter(item => {
+      return item.path === path;
+    });
+
+    return check;
+  }
+
+  globalChecks(): ObjectCheck[] {
+    return this.checkForPath('');
+  }
+
+  message(): string {
+    const checks = this.globalChecks();
+    let message = checks.map(check => check.message).join(',');
+
+    if (!message.length && !this.isValid()) {
+      message = 'Not all checks passed!';
+    }
+
+    return message;
+  }
+
+  render() {
+    const validation = this.validation();
+    const IconComponent = validation.icon;
+
+    return (
+      <div>
+        <p style={{ color: validation.color }}>
+          <IconComponent /> {this.message()}
+        </p>
+      </div>
+    );
```

Comment:
Replace 
```ts
<div> </div>

``` by
```ts
 <></>
```

about IconComponent you can do 
```ts
{validation.icon}
```

to render the icon instead create a new var